src/early-types.lisp

```diff
@@ -164,6 +164,14 @@ Signals an error if FOREIGN-TYPE is undefined."))
   (:documentation
    "Return the size in bytes of a foreign type."))
 
+(define-compiler-macro foreign-type-size (&whole form foreign-type)
+  ;; Don't pass &environment to CONSTANTP or MACROEXPAND as it may make
+  ;; surprising behavior with LOAD-TIME-VALUE.
```

Comment:
Could you provide more examples of this?

Reply:
I rephrased this now that I took a second look at this. Clarified: since L-T-V executes under a null lexical environment, we shouldn't be checking if the foreign type is constant under &environment.